HeliPort/AppDelegate.swift

```diff
@@ -38,38 +38,59 @@ class AppDelegate: NSObject, NSApplicationDelegate, NSMenuDelegate {
         let interface = String(cString: &drv_info.bsd_name.0)
         guard !version.isEmpty, !interface.isEmpty else {
             Log.error("itlwm kext not loaded!")
+            #if !DEBUG
+                Alert(check: "itlwm")
+            #endif
             return false
         }
 
         Log.debug("Loaded itlwm \(version) as \(interface)")
-
         return true
     }
 
     private func checkAPI() {
 
+        // It's fine for users to bypass this check by launching HeliPort first then loading itlwm in terminal
+        // Only advanced users do so, and they know what they are doing
         guard checkDriver(), IOCTL_VERSION != drv_info.version else {
             return
         }
-
-        let verAlert = NSAlert()
-        verAlert.alertStyle = .critical
-        verAlert.messageText = NSLocalizedString("itlwm Version Mismatch", comment: "")
-        verAlert.informativeText =
-            NSLocalizedString("HeliPort API Version: ", comment: "") + String(IOCTL_VERSION) +
-            "\n" +
-            NSLocalizedString("itlwm API Version: ", comment: "") + String(drv_info.version)
-        verAlert.addButton(
-            withTitle: NSLocalizedString("Visit OpenIntelWireless on GitHub", comment: "")
-        ).keyEquivalent = "\r"
-        verAlert.addButton(withTitle: NSLocalizedString("Quit HeliPort", comment: ""))
-
-        NSApplication.shared.activate(ignoringOtherApps: true)
-
-        if verAlert.runModal() == .alertFirstButtonReturn {
-            NSWorkspace.shared.open(URL(string: "https://github.com/OpenIntelWireless")!)
-        }
-
-        NSApp.terminate(nil)
+        Alert(check: "Mismatch")
+    }
+    
+    private func Alert(check: String) {
+            let verAlert = NSAlert()
+            verAlert.alertStyle = .critical
+        
+            if check == "Mismatch" {
+                verAlert.messageText = NSLocalizedString("itlwm Version Mismatch", comment: "")
+                verAlert.informativeText =
+                NSLocalizedString("HeliPort API Version: ", comment: "") + String(IOCTL_VERSION) +
+                "\n" +
+                NSLocalizedString("itlwm API Version: ", comment: "") + String(drv_info.version)
+                verAlert.addButton(withTitle: NSLocalizedString("Quit HeliPort", comment: "")).keyEquivalent = "\r"
+            } else {
+                verAlert.messageText = NSLocalizedString("itlwm is not running", comment: "")
+                    verAlert.informativeText = NSLocalizedString("Install or load itlwm v.", comment: "") + String(IOCTL_VERSION)
```

Comment:
Please remove this line, `IOCTL_VERSION` is Heliport's API but not itlwm's `MARKETING_VERSION`

Reply:
Done